payments-core/src/main/java/com/stripe/android/view/CardBrandViewRebuild.kt

```diff
@@ -0,0 +1,249 @@
+        set(value) {
+            state = state.copy(merchantPreferredNetworks = value)
+        }
+
+    var shouldShowCvc: Boolean
+        get() = state.shouldShowCvc
+        set(value) {
+            state = state.copy(shouldShowCvc = value)
+        }
+
+    var shouldShowErrorIcon: Boolean
+        get() = state.shouldShowErrorIcon
+        set(value) {
+            state = state.copy(shouldShowErrorIcon = value)
+        }
+
+    internal var tintColorInt: Int
+        get() = state.tintColor
+        set(value) {
+            state = state.copy(tintColor = value)
+        }
+
+    init {
+        isClickable = false
+        isFocusable = false
+        iconView.setBackgroundResource(com.stripe.payments.model.R.drawable.stripe_ic_unknown)
+
+        determineCardBrandToDisplay(
+            state.userSelectedBrand,
+            state.brand,
+            state.possibleBrands,
+            state.merchantPreferredNetworks
+        )
+        updateBrandSpinner()
+    }
+
+    fun createNetworksParam(): PaymentMethodCreateParams.Card.Networks? {
+        return PaymentMethodCreateParams.Card.Networks(
+            preferred = brand.takeIf { it != Unknown }?.code,
+        ).takeIf {
+            isCbcEligible && possibleBrands.size > 1
+        }
+    }
+
+    private fun handleBrandSelected(brand: CardBrand?) {
+        brand?.let {
+            state = state.copy(userSelectedBrand = brand)
+        }
+    }
+
+    private fun determineCardBrandToDisplay(
+        userSelectedBrand: CardBrand?,
+        autoDeterminedBrand: CardBrand,
+        possibleBrands: List<CardBrand>,
+        merchantPreferredBrands: List<CardBrand>,
+    ) {
+        brand = if (possibleBrands.size > 1) {
+            selectCardBrandToDisplay(userSelectedBrand, possibleBrands, merchantPreferredBrands)
+        } else {
+            autoDeterminedBrand
+        }
+    }
+
+    override fun onSaveInstanceState(): Parcelable {
+        val superState = super.onSaveInstanceState()
+        return SavedState(superState, state)
+    }
+
+    override fun onRestoreInstanceState(state: Parcelable?) {
+        val savedState = state as? SavedState
+        this.state = savedState?.state ?: State()
+        determineCardBrandToDisplay(
+            this.state.userSelectedBrand,
+            this.state.brand,
+            this.state.possibleBrands,
+            this.state.merchantPreferredNetworks
+        )
+        updateBrandSpinner()
+        super.onRestoreInstanceState(savedState?.superState)
+    }
+
+    @Parcelize
+    internal data class SavedState(
+        val superSavedState: Parcelable?,
+        val state: State,
+    ) : BaseSavedState(superSavedState), Parcelable
+
+    private fun updateBrandSpinner() {
+        chevron.visibility = if (possibleBrands.size > 1)  View.VISIBLE else View.GONE
+        if (possibleBrands.size > 1) {
+            val adapter = BrandAdapter(context, possibleBrands)
+            this.setOnClickListener { listPopup.show() }
```

Comment:
I feel like tapping the card brand icon when the dropdown is open should dismiss it. Right now, it triggers another open animation.

Reply:
Updated 👍